### ISSUE TITLE
Window Freezes During Toast

### ISSUE THREAD

**Issue by VisenDev** [david-vanderson/dvui#80]
Using the toast widget in the demo window causes the entire window to freeze for a few seconds.

If this is intended behavior, it should probably be adjusted because it feels bad when the application goes non-responsive as a user.

Platform: x86_64 Linux
Backend: SDL2

The same behavior also showed on my mac - the OS even gave me a spinning mouse pointer animation to show that the application was not responding.

**Comment by david-vanderson** [comment]
What version of SDL do you have?  There was a bug a while back, maybe I need to put the mitigation back in?

**Comment by david-vanderson** [comment]
Does this fix it for you?

```zig
diff --git a/src/backends/SDLBackend.zig b/src/backends/SDLBackend.zig
index 184254e..7ee3c00 100644
--- a/src/backends/SDLBackend.zig
+++ b/src/backends/SDLBackend.zig
@@ -228,6 +228,11 @@ pub fn waitEventTimeout(_: *SDLBackend, timeout_micros: u32) void {
         // wait with a timeout
         const timeout = @min((timeout_micros + 999) / 1000, std.math.maxInt(c_int));
         _ = c.SDL_WaitEventTimeout(null, @as(c_int, @intCast(timeout)));
+
+        // TODO: this call to SDL_PollEvent can be removed after resolution of
+        // https://github.com/libsdl-org/SDL/issues/6539
+        // maintaining this a little longer for people with older SDL versions
+        _ = c.SDL_PollEvent(null);
     } else {
         // don't wait
     }
```

**Comment by VisenDev** [comment]
> Does this fix it for you?

Yes making those changes in the backend file fixes the bug

My `yum` installed SDL2 is the following

```
Name         : SDL2
Version      : 2.26.5
Release      : 2.fc39
Architecture : x86_64
Size         : 7.6 M
Source       : None
Repository   : nobara-baseos-multilib-39
Summary      : Cross-platform multimedia library
URL          : http://www.libsdl.org/
License      : Zlib AND MIT AND Apache-2.0 AND (Apache-2.0 OR MIT)
Description  : Simple DirectMedia Layer (SDL) is a cross-platform multimedia library designed
             : to provide fast access to the graphics frame buffer and audio device.
```  
             
```         
Name         : SDL2-devel
Version      : 2.26.5
Release      : 2.fc39
Architecture : x86_64
Size         : 381 k
Source       : SDL2-2.26.5-2.fc39.src.rpm
Repository   : nobara-baseos-multilib-39
Summary      : Files needed to develop Simple DirectMedia Layer applications
URL          : http://www.libsdl.org/
License      : Zlib AND MIT AND Apache-2.0 AND (Apache-2.0 OR MIT)
Description  : Simple DirectMedia Layer (SDL) is a cross-platform multimedia library designed
             : to provide fast access to the graphics frame buffer and audio device. This
             : package provides the libraries, include files, and other resources needed for
             : developing SDL applications.



```

**Comment by david-vanderson** [comment]
Okay yes the SDL fix went in to 2.28 - I'll push this workaround back in.  Thanks!

**Comment by david-vanderson** [comment]
I've pushed the workaround back in.  Let me know if it doesn't work.
